src/background-job/upload_to_geonode/upload_to_geonode_job.py

```diff
@@ -0,0 +1,176 @@
+import os
+import json
+import requests
+
+geonode_url = os.getenv("GEONODE_URL")
+username = os.getenv("GEONODE_USERNAME")
+password = os.getenv("GEONODE_PASSWORD")
+dataset_path = "../../output_data/GeoTiffs"
+dataset_type = ".tif"
+
+
+def get_categories(api_url, file_name="geonode_category.json"):
+    try:
+        # Fetch data from the API
+        response = requests.get(api_url)
+        # Will raise an error for 4xx or 5xx status codes
+        response.raise_for_status()
+
+        selected_categories = [
+            "cdi-raster-map",
+            "spi-raster-map",
+            "ndvi-raster-map",
+            "lst-raster-map"
+        ]
+        categories = response.json().get("objects")
+        categories = list(filter(
+            lambda x: x["identifier"] in selected_categories,
+            categories
+        ))
+
+    except requests.exceptions.RequestException as e:
+        print(f"Error fetching data from API: {e}")
+        return {}
+
+    # Convert list to dictionary
+    category = {c['identifier'].split('-')[0]: c['id'] for c in categories}
+
+    # Check if the file exists
+    if not os.path.exists(file_name):
+        # Write dictionary to JSON file
+        with open(file_name, 'w') as json_file:
+            json.dump(category, json_file, indent=4)
+
+    # Read and return the contents of the JSON file
+    with open(file_name, 'r') as json_file:
+        return json.load(json_file)
+
+
+def write_failure_message(response):
+    print("Request failed.")
+    print("Status Code:", response.status_code)
+    print("Response:", response.text)
+
+
+def upload_to_geonode(base_file_path, xml_file_path=None, sld_file_path=None):
+    api_url = f"{geonode_url}api/v2/uploads/upload?format=json"
+    files = {
+        "base_file": open(base_file_path, "rb"),
+    }
+    if xml_file_path:
+        files["xml_file"] = open(xml_file_path, "rb")
+    if sld_file_path:
+        files["sld_file"] = open(sld_file_path, "rb")
+    data = {
+        "store_spatial_files": False,
+        "overwrite_existing_layer": True,
+        "skip_existing_layers": False,
+    }
+    response = requests.post(
+        api_url,
+        auth=(username, password),
+        files=files,
+        data=data
+    )
+    if response.status_code == 201:
+        json_response = response.json()
+        return json_response.get("execution_id")
+    else:
+        write_failure_message(response)
+        return None
```

Comment:
Any reason not to just throw a `RuntimeError` and exit here as in `tracking_upload_progress`? I guess not every status code that isn't 201 is an error code but then perhaps `write_failure_message` is a misleading name?